tikv-client/src/main/java/com/pingcap/tikv/types/JsonType.java

```diff
@@ -57,171 +52,135 @@ public JsonType(MySQLType type, int flag, int len, int decimal, String charset,
-      cdi.readFully(readBuffer, off, len);
-    } catch (IOException e) {
-      throw new UncheckedIOException(e);
-    }
+  private long parseInt64(CodecDataInput cdi) {
+    return cdi.readLongLSB();
   }
 
-  private long parseInt64(DataInput cdi) {
-    byte[] readBuffer = new byte[8];
-    readFully(cdi, readBuffer);
-    return ((long) readBuffer[7] << 56)
-        + ((long) (readBuffer[6] & 255) << 48)
-        + ((long) (readBuffer[5] & 255) << 40)
-        + ((long) (readBuffer[4] & 255) << 32)
-        + ((long) (readBuffer[3] & 255) << 24)
-        + ((readBuffer[2] & 255) << 16)
-        + ((readBuffer[1] & 255) << 8)
-        + ((readBuffer[0] & 255) << 0);
+  private int parseUint32(CodecDataInput cdi) {
+    return cdi.readIntLSB();
   }
 
-  private long parseUint32(DataInput cdi) {
-    byte[] readBuffer = new byte[4];
-    readFully(cdi, readBuffer);
-
-    return ((long) (readBuffer[3] & 255) << 24)
-        + ((readBuffer[2] & 255) << 16)
-        + ((readBuffer[1] & 255) << 8)
-        + ((readBuffer[0] & 255) << 0);
+  private int parseUint16(CodecDataInput cdi) {
+    return cdi.readUnsignedShortLSB();
   }
 
-  private double parseDouble(DataInput cdi) {
-    byte[] readBuffer = new byte[8];
-    readFully(cdi, readBuffer);
-    return Double.longBitsToDouble(
-        ((long) readBuffer[7] << 56)
-            + ((long) (readBuffer[6] & 255) << 48)
-            + ((long) (readBuffer[5] & 255) << 40)
-            + ((long) (readBuffer[4] & 255) << 32)
-            + ((long) (readBuffer[3] & 255) << 24)
-            + ((readBuffer[2] & 255) << 16)
-            + ((readBuffer[1] & 255) << 8)
-            + ((readBuffer[0] & 255) << 0));
+  private double parseDouble(CodecDataInput cdi) {
+    return cdi.readDoubleLSB();
   }
 
-  private int parseUint16(DataInput cdi) {
-    byte[] readBuffer = new byte[2];
-    readFully(cdi, readBuffer);
-
-    return ((readBuffer[1] & 255) << 8) + ((readBuffer[0] & 255) << 0);
-  }
-
-  private String parseString(DataInput di, long length) {
-
-    byte[] buffer = new byte[Math.toIntExact(length)];
-    readFully(di, buffer);
+  private String parseString(CodecDataInput cdi, int length) {
+    byte[] buffer = new byte[length];
+    cdi.readFully(buffer, 0, length);
     return new String(buffer, Charsets.UTF_8);
   }
 
   /**
    * func Uvarint(buf []byte) (uint64, int) { var x uint64 var s uint for i, b := range buf { if b <
    * 0x80 { if i > 9 || i == 9 && b > 1 { return 0, -(i + 1) // overflow } return x | uint64(b)<<s,
    * i + 1 * } x |= uint64(b&0x7f) << s s += 7 } return 0, 0 }
-   *
-   * @param di
-   * @return
    */
-  private long parseDataLength(DataInput di) {
-    long x = 0;
+  private int parseDataLength(CodecDataInput cdi) {
+    int x = 0;
     byte b;
     int i = 0;
     int s = 0;
-    while ((b = readByte(di)) < 0) {
+    while ((b = cdi.readByte()) < 0) {
       if (i == 9) {
         throw new IllegalArgumentException("overflow: found >=9 leading bytes");
```

Comment:
"overflow: found >=9 leading bytes" I am a little bit confused by the error msg.

Reply:
I think this logic was copied from TiDB.